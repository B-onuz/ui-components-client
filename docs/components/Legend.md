# Legend (Component)

Componente Legend utilizado em conjunto com o componente `Fieldset`.
Usado para dar um nome para uma sessão do formulário.

## Como utilizar

```js
import Legend from 'ui-components/lib/components/Legend'
```

```jsx
<Legend>Foo bar</Legend>
```
