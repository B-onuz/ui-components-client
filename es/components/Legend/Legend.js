"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n  font-family: inherit;\n  font-weight: 600;\n  margin-bottom: 16px;\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteralLoose(strings, raw) { if (!raw) { raw = strings.slice(0); } strings.raw = raw; return strings; }

var Legend = _styledComponents["default"].legend(_templateObject());

Legend.displayName = 'Legend';
var _default = Legend;
exports["default"] = _default;