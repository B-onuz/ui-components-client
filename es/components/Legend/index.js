"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _Legend = _interopRequireDefault(require("./Legend"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var _default = _Legend["default"];
exports["default"] = _default;