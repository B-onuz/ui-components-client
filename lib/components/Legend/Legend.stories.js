'use strict'

exports.__esModule = true
exports.LegendExample = exports['default'] = void 0

var _react = _interopRequireDefault(require('react'))

var _Legend = _interopRequireDefault(require('./Legend'))

var _this = void 0,
  _jsxFileName = '/home/junior/Fontes/ui-components-client/src/components/Legend/Legend.stories.js'

function _interopRequireDefault(obj) {
  return obj && obj.__esModule ? obj : { default: obj }
}

var _default = {
  title: 'Legend',
  component: _Legend['default'],
}
exports['default'] = _default

var LegendExample = function LegendExample() {
  return /*#__PURE__*/ _react['default'].createElement(
    _Legend['default'],
    {
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 9,
        columnNumber: 36,
      },
    },
    'Nome completo'
  )
}

exports.LegendExample = LegendExample
