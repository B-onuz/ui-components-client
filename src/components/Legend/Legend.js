import React from 'react'
import styled from 'styled-components'

const Legend = styled.legend`
  font-family: inherit;
  font-weight: 600;
  margin-bottom: 16px;
`

Legend.displayName = 'Legend'

export default Legend
