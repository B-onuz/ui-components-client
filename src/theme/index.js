const colors = {
  red: '#db2828',
  lightRed: '#ef4d5a',
  mediumRed: '#FF4F5D',
  darkRed: '#9c2543',
  fullDarkRed: '#550e33',
  orange: '#f2711c',
  yellow: '#fbbd08',
  olive: '#b5cc18',
  green: '#21BA45',
  teal: '#00b5ad',
  blue: '#2185d0',
  facebookBlue: '#3B5998',
  blueLink: '#4285F4',
  violet: '#6435c9',
  purple: '#a333c8',
  pink: '#e03997',
  brown: '#a5673f',
  grey: '#767676',
  textGrey: '#8D8D8D',
  lightGrey: '#6D6D6D',
  mediumGrey: '#838383',
  darkGrey: '#9A9A9A',
  boldGray: '#55555A',
  black: '#1b1c1d',
  white: '#fdfdfd',
  warning: '#7A4D05',
  success: '#1E561F',
  info: '#0E566C',
  danger: '#973937',
  cyan: '#1CB5AD',
  disabledGrey: '#ffffff1f',
  disabledGreyText: '#ffffff4d',
}

colors.primary = colors.lightRed
colors.secondary = colors.fullDarkRed
colors.default = '#e0e1e2'

const breakpoints = ['40em', '52em', '64em', '80em']

breakpoints.sm = breakpoints[0]
breakpoints.md = breakpoints[1]
breakpoints.lg = breakpoints[2]
breakpoints.xl = breakpoints[3]

const fontSizes = [12, 14, 16, 20, 24, 32]

export default {
  fontFamily: `'Lato', sans-serif`,
  colors,
  fontSizes,
  breakpoints,
}
