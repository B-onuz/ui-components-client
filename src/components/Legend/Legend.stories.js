import React from 'react'
import Legend from './Legend'

export default {
  title: 'Legend',
  component: Legend,
}

export const LegendExample = () => <Legend>Nome completo</Legend>
