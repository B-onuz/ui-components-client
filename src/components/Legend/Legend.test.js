import React from 'react'
import Legend from './Legend.js'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import theme from '../../theme'
import 'jest-styled-components'

test('Legend changes the class when hovered', () => {
  const component = renderer.create(
    <ThemeProvider theme={theme}>
      <Legend />
    </ThemeProvider>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
