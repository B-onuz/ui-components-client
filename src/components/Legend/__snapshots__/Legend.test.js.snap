// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Legend changes the class when hovered 1`] = `
.c0 {
  font-family: inherit;
  font-weight: 600;
  margin-bottom: 16px;
}

<legend
  className="c0"
/>
`;
